cowprotocol/cowswap
# feat(hook-store): override Hook Gas limit and tokens balances with Tenderly Simulation Data

# Summary

Using `gasUsed` data of Tenderly simulation (added on bff on this [PR](https://github.com/cowprotocol/bff/pull/93)) to override `gasLimit` hook data.

# To Test

1. Run the `bff` server locally in this [PR](https://github.com/cowprotocol/bff/pull/93) context
2. Run the app locally connecting it with your local `bff`
3. Use the Build Hook dApp to create one simple hook with a low gas limit. The data that I used to test was a pre-hook with:
- Target: 0xfFf9976782d46CC05630D1f6eBAb18b2324d6B14
- Gas Limit: 10
- Calldata: 0x
4. Select two tokens go until the swap review and check the gas limit. Check if the gas limit is different from the one that you inputted and 10% bigger than the simulated one (click on the Tenderly simulation link to check the simulated)
<img width="327" alt="image" src="https://github.com/user-attachments/assets/5d8cf9e5-71a7-449e-9583-dffbce2a326d">


Note: The PR is in draft until the [balanceDiff PR](https://github.com/cowprotocol/cowswap/pull/5027) is merged. But feel free to test it (only the diffs will have that context as well)
